extensions/blocks/donations/amount-input.js

```diff
@@ -0,0 +1,96 @@
+/**
+ * External dependencies
+ */
+import { CURRENCIES } from '@automattic/format-currency';
+
+/**
+ * WordPress dependencies
+ */
+import { RichText } from '@wordpress/block-editor';
+
+/**
+ * Internal dependencies
+ */
+import { minimumTransactionAmountForCurrency } from '../../shared/currencies';
+
+const parseAmount = ( amount, currency ) => {
```

Comment:
If this gets complicated, it wouldn't be unreasonable to let the server do the heavy lifting here. Note that there will be corner cases with negative values, random characters, and so on. Unit tests will be helpful for both a client/server library

Reply:
> let the server do the heavy lifting here

Do you mean with an API call every time an amount changes? Or as part of the server render callback?

A new API endpoint seems a bit complex at this time, and while we still need to validate the amounts in the server render callback (a user could enter anything when in the code editor mode which will bypass any client check), not including any validation in the client will leave users unaware of invalid amounts and might get confused when they see the published post does not contain the amount they entered.